etl/steps/data/garden/emdat/2022-11-24/shared.py

```diff
@@ -0,0 +1,591 @@
+import datetime
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
+
+import numpy as np
+import pandas as pd
+from owid import catalog
+from owid.datautils import geo
+from structlog import get_logger
+
+from etl.paths import DATA_DIR
+
+CURRENT_DIR = Path(__file__).parent
+
+log = get_logger()
+
+# Aggregate regions to add, following OWID definitions.
+# Regions and income groups to create by aggregating contributions from member countries.
+# In the following dictionary, if nothing is stated, the region is supposed to be a default continent/income group.
+# Otherwise, the dictionary can have "regions_included", "regions_excluded", "countries_included", and
+# "countries_excluded". The aggregates will be calculated on the resulting countries.
+REGIONS = {
+    # Default continents.
+    "Africa": {},
+    "Asia": {},
+    "Europe": {},
+    "European Union (27)": {},
+    "North America": {},
+    "Oceania": {},
+    "South America": {},
+    "World": {},
+    # Income groups.
+    "Low-income countries": {},
+    "Upper-middle-income countries": {},
+    "Lower-middle-income countries": {},
+    "High-income countries": {},
+}
+
+# When creating region aggregates, decide how to distribute historical regions.
+# The following decisions are based on the current location of the countries that succeeded the region, and their income
+# group. Continent and income group assigned corresponds to the continent and income group of the majority of the
+# population in the member countries.
+HISTORIC_TO_CURRENT_REGION: Dict[str, Dict[str, Union[str, List[str]]]] = {
+    "Czechoslovakia": {
+        "continent": "Europe",
+        "income_group": "High-income countries",
+        "regions_included": [
+            # Europe - High-income countries.
+            "Czechia",
+            "Slovakia",
+        ],
+    },
+    "East Germany": {
+        "continent": "Europe",
+        "income_group": "",
+        "regions_included": [
+            # Europe - High-income countries.
+            "Germany",
+        ],
+    },
+    "West Germany": {
+        "continent": "Europe",
+        "income_group": "",
+        "regions_included": [
+            # Europe - High-income countries.
+            "Germany",
+        ],
+    },
+    "Netherlands Antilles": {
+        "continent": "North America",
+        "income_group": "High-income countries",
+        "regions_included": [
+            # North America - High-income countries.
+            "Aruba",
+            "Curacao",
+            "Sint Maarten (Dutch part)",
+            "Caribbean Netherlands",
```

Comment:
True! That conversation got a bit unresolved, since it wasn't clear to me if it was a good idea to remove "Caribbean Netherlands" from countries_regions. In any case neither of them are present in this dataset, so I'll change it for consistency, but it doesn't affect the data.